docs/sphinxext/mantiddoc/directives/properties.py

```diff
@@ -75,11 +75,28 @@ def _create_properties_table(self):
 
             for prop in alg_properties:
                 # Append a tuple of properties to the list.
+                # if type is not standard, it should be enumerated string property and should be considered as a string
+                str_prop_type = str(prop.type)
+                if str_prop_type not in [
```

Comment:
I think a simpler version of this would be
```python
str_prop_type = property_type_dict.get(str_prop_type, str_prop_type)
if "EnumeratedString" in str_prop_type and "Mantid" in str_prop_type:
    str_prop_type = "string"
```
would be more robust because new types would not necessarily have to be added for it to keep working. The current version will fail in subtle ways when new types appear and will be hard to debug.

Reply:
Yes current version should be synchronized with Property.cpp

Reply:
My suggestion means you don't have to update. Look for what is acceptable rather than have an exhaustive list of what is unacceptable. Build the docs and look at Rebin's docs. I'm suspicious you'll see different behavior than you expect.